src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProjectFactory.cs

```diff
@@ -63,8 +64,6 @@ IWorkspaceProjectContext IWorkspaceProjectContextFactory.CreateProjectContext(st
             string? assemblyName,
             CancellationToken cancellationToken)
         {
-            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
📝  This is a NOP when the code is already on the main thread, and the recommended pattern is to place the explicit switch in all asynchronous methods that are expected to run on the main thread.

Reply:
So this code does *not* need the UI thread, except for the F# hack when it's loading the package. So the goal here is to target that specifically.

Reply:
Do you have a different pattern to recommend? I'm unclear what your proposal is here.

Reply:
> So this code does not need the UI thread, except for the F# hack when it's loading the package. So the goal here is to target that specifically.

That works 👍

Reply:
And I agree this isn't pretty, but the hack is kept in a neat and tidy single code block, so I'm aiming for that. I have a bug with the F# team to follow up on whether this is necessary anymore.